src/contracts/RichText/TextExtractor/NodeFilterInterface.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Contracts\FieldTypeRichText\RichText\TextExtractor;
+
+use DOMNode;
+
+/**
+ * Filters nodes for text extraction.
+ */
+interface NodeFilterInterface
+{
+    /**
+     * Return false to preserve the node, true to remove it.
+     */
+    public function filter(DOMNode $node): bool;
```

Comment:
IMHO the result of false/true combined with `filter` might be a bit misleading. My first thought was that it should work the opposite way. Maybe changing it to `filterOut` would be more clear?

Reply:
I personalny stand with the original naming.

Reply:
I have to agree with @alongosz here.

Most methods ([`ArrayCollection::filter`](https://github.com/doctrine/collections/blob/d8af7f248c74f195f7347424600fd9e17b57af59/src/ArrayCollection.php#L379-L382)) and functions ([`array_filter`](https://www.php.net/manual/en/function.array-filter.php)) work in the opposite way:
 * when `true`, an entry is preserved
 * when `false`, an entry is removed

Therefore, I'd suggest reversing the logic to comply with generally established PHP practice.